records/state/state.go

```diff
@@ -103,7 +116,18 @@ func (t *Task) containerIP(src string) string {
 			continue
 		}
 
-		// find the latest docker-inspect label
+		// first try to extract the address from the NetworkInfo structure
+		if len(status.ContainerStatus.NetworkInfos) > 0 {
+			// TODO(CD): handle multiple NetworkInfo objects
+			// TODO(CD): only create A records if the address is IPv4
+			// TODO(CD): create AAAA records if the address is IPv6
+			latestContainerIP = status.ContainerStatus.NetworkInfos[0].IPAddress
+			latestTimestamp = status.Timestamp
+			break
```

Comment:
might be breaking too early here if there are multiple TASK_RUNNING status entries and they're out of order w/ respect to timestamps (this break has a different effect than the one in the label-driven loop below)